packages/connection-form/src/components/connection-form.tsx

```diff
@@ -97,6 +108,165 @@ type ConnectionFormPropsWithoutPreferences = {
   onSaveConnectionClicked: (connectionInfo: ConnectionInfo) => Promise<void>;
 };
 
+const colorPreviewStyles = css({
+  height: '16px',
+  width: '16px',
+  marginRight: spacing[2],
+});
+
+const ColorCircleGlyph = createGlyphComponent('ColorCircle', (props) => (
```

Comment:
If you are spreading props on the component, you should pick up your extra props to avoid spreading them in an unexpected place

```suggestion
const ColorCircleGlyph = createGlyphComponent('ColorCircle', ({ hexColor, ...props }) => (
```

Reply:
Fixed! I had a few issues with the function as the typings are a little bit cryptic for me, but with your suggestion and using the react intrinsic props and a union type now it compiles properly.